File: testdata/fixtures/javascript/md5-used-as-password.js
const crypto = require('crypto');

function hashPassword(password) {
  const h = crypto.createHash('md5');
  h.update(password);
  // ruleid: md5-used-as-password
  return storePassword(h.digest('hex'));
}

function savePassword(password) {
  const h = crypto.createHash('md5');
  h.update(password);
  // ruleid: md5-used-as-password
  return savePasswordHash(h.digest('hex'));
}

function hashFile(path) {
  const h = crypto.createHash('md5');
  h.update(path);
  // ok: md5-used-as-password
  return h.digest('hex');
}

function hashPasswordSha(password) {
  const h = crypto.createHash('sha256');
  h.update(password);
  // ok: md5-used-as-password
  return storePassword(h.digest('hex'));
}

File: testdata/fixtures/ruby/md5-used-as-password.rb
require 'digest'

def hash_login(pw)
  h = Digest::MD5.hexdigest(pw)
  # ruleid: md5-used-as-password
  save_password(h)
end

def hash_register(pw)
  h = Digest::MD5.new
  h.update(pw)
  # ruleid: md5-used-as-password
  update_password(h.hexdigest)
end

def hash_reset(pw)
  d = Digest::MD5.digest(pw)
  # ruleid: md5-used-as-password
  store_password(d)
end

def hash_sha(pw)
  h = Digest::SHA256.hexdigest(pw)
  # ok: md5-used-as-password
  save_password(h)
end

def digest_only(data)
  # ok: md5-used-as-password
  Digest::MD5.hexdigest(data)
end

def plain(pw)
  # ok: md5-used-as-password
  save_password(pw)
end
